osidb/models.py

```diff
@@ -809,6 +810,35 @@ def _validate_embargoing_public_flaw(self):
         if not old_flaw.embargoed and self.is_embargoed:
             raise ValidationError("Embargoing a public flaw is futile")
 
+    def _validate_cwe_format(self):
+        """
+        Check if CWE string is well formated
+        """
+        cwe_data = self.cwe_id
+        # First, remove possible [auto] suffix from the CWE entry
+        # [auto] suffix means value was assigned by a script during mass update
+        if len(cwe_data) > 6 and cwe_data.endswith("[auto]"):
+            cwe_data = cwe_data[:-6]
+
+        # Then split data on arrows ->, later we will parse the elements separately
+        arrow_count = len(re.findall("->", cwe_data))
+        parsed_elements = list(filter(None, cwe_data.split("->")))
+
+        # Ensure number of elements is one bigger then count of arrows, to catch
+        # stuff like: CWE-123->
+        if len(parsed_elements) != (arrow_count + 1):
```

Comment:
CWE is not a mandatory flaw attribute. You can see at https://git.prodsec.redhat.com/devops/sfm2/-/blob/master/sfm2/validation/flaw.py#L2035 that for an empty CWE string we do not perform the validation in SFM2. It seems to me that with an empty CWE we would however fail here. Please check it and eventually fix